python/ray/data/_internal/arrow_block.py

```diff
@@ -224,7 +228,7 @@ def schema(self) -> "pyarrow.lib.Schema":
     def to_pandas(self) -> "pandas.DataFrame":
         from ray.air.util.data_batch_conversion import _cast_tensor_columns_to_ndarrays
 
-        df = self._table.to_pandas()
+        df = self._table.to_pandas(use_threads=False)
```

Comment:
Any particular reason for this / document the reason for this? Naively it seems like we'd want to use threads.

Reply:
@ericl As mentioned in standup and in the PR description, the plan is to decompose this PR into a stack, this PR is just the end-to-end mono-PR used for making sure that everything works and for benchmarking the end-state.

Reply:
I didn't notice having any positive or negative effect on my benchmarking after disabling multithreaded Pandas conversion, so I thought I'd set this to `False` similar to how we disable multi-threaded Arrow I/O, and only enable it if it showed speedups in the benchmarking.

I think I'll probably set this back to the default for now when I split out the PRs, though.